managed/services/telemetry/distribution_util_test.go

```diff
@@ -0,0 +1,231 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package telemetry
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	pmmv1 "github.com/percona-platform/saas/gen/telemetry/events/pmm"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/percona/pmm/api/serverpb"
+)
+
+func Test_distributionUtilServiceImpl_getDistributionMethodAndOS(t *testing.T) {
+	const (
+		tmpDistributionFilePathPrefix = "/tmp/distribution"
+		tmpOsInfoFilePathPrefix       = "/tmp/version"
+
+		ami          = "ami"
+		ovf          = "ovf"
+		azure        = "azure"
+		digitalocean = "digitalocean"
+		docker       = "docker"
+
+		DistributionMethodDistributionMethodInvalid = 0
+		DistributionMethodDOCKER                    = 1
+		DistributionMethodOVF                       = 2
+		DistributionMethodAMI                       = 3
+		DistributionMethodAZURE                     = 4
+		DistributionMethodDO                        = 5
+
+		PmmV1DistributionMethodDistributionMethodInvalid = 0
+		PmmV1DistributionMethodDOCKER                    = 1
+		PmmV1DistributionMethodOVF                       = 2
+		PmmV1DistributionMethodAMI                       = 3
+		PmmV1DistributionMethodAZURE                     = 4
+		PmmV1DistributionMethodDO                        = 5
```

Comment:
Why not to use `serverpb.DistributionMethod_*` and `pmmv1.DistributionMethod_*` values directly?

Reply:
Better to not use constants from the source code, because in theory they accidentally by mistake could be changed, and in that case we won't find the error.

Reply:
These values are part of API, and we use protobuf descriptors there to detect breaking changes. I think you can skip this kind of checks here. But it's up to you.